File: Mobile MediaPlayer (Xamarin Android)/MediaPlayer/MediaPlayer/Core/NotificationBroadcast.cs
using Android.Content;
using Android.Support.V7.App;
using Android.Widget;

namespace MediaPlayer.Core
{
    public class NotificationBroadcast : BroadcastReceiver
    {
        public enum Actions { shuffleAction, previousAction, playPauseAction, nextAction }

        MainActivity MainActivity;

        public NotificationBroadcast(MainActivity activity)
        {
            MainActivity = activity;
        }

        public override void OnReceive(Context context, Intent intent)
        {
            if(intent.Action == Actions.shuffleAction.ToString())
            {
                MainActivity.ChangePlayingMode();
            }
            else if (intent.Action == Actions.previousAction.ToString())
            {
                MainActivity.ChangeToPreviousSong();
            }
            else if (intent.Action == Actions.playPauseAction.ToString())
            {
                MainActivity.PlayPauseSong();
            }
            else if (intent.Action == Actions.nextAction.ToString())
            {
                MainActivity.ChangeToNextSong();
            }
        }
    }
}

File: Mobile MediaPlayer (Java Android)/app/src/main/java/com/example/mediaplayer/Core/NotificationBroadcast.java
package com.example.mediaplayer.Core;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;

import com.example.mediaplayer.MainActivity;

public class NotificationBroadcast extends BroadcastReceiver {
    public enum Actions { shuffleAction, previousAction, playPauseAction, nextAction }

    com.example.mediaplayer.MainActivity MainActivity;

    public NotificationBroadcast() {}

    public NotificationBroadcast(MainActivity activity)
    {
        MainActivity = activity;
    }

    @Override
    public void onReceive(Context context, Intent intent)
    {
        if(intent.getAction().equals(Actions.shuffleAction.toString()))
        {
            MainActivity.ChangePlayingMode();
        }
        else if (intent.getAction().equals(Actions.previousAction.toString()))
        {
            MainActivity.ChangeToPreviousSong();
        }
        else if (intent.getAction().equals(Actions.playPauseAction.toString()))
        {
            MainActivity.PlayPauseSong();
        }
        else if (intent.getAction().equals(Actions.nextAction.toString()))
        {
            MainActivity.ChangeToNextSong();
        }
        else if(intent.getAction().equals(Intent.ACTION_HEADSET_PLUG))
        {
            switch(intent.getIntExtra("state", -1)) {
                case(0):
                    MainActivity.PauseSong();
                    break;
                case(1):
                    MainActivity.PlaySong();
                    break;
            }
        }
    }
}
